src/main/scala/tech/cryptonomic/conseil/generic/rpc/RpcHandler.scala

```diff
@@ -0,0 +1,90 @@
+package tech.cryptonomic.conseil.generic.rpc
+
+import cats._
+import cats.arrow._
+import cats.data.Kleisli
+
+/** Defines operations to execute calls against a remote node
+  *
+  * For the sake of comprehension, we can compare a Kleisli[F, A, B] data-type
+  * as a "special" function that wraps the result in some "effect" F, i.e. A => F[B]
+  *
+  * @tparam Eff the effect associated with execution (Future, IO, Either, Id...)
+  * @tparam Command represents the input to identify the endpoint (e.g. a path string)
+  * @tparam Reponse represents the output from the node (e.g. a json string)
+  */
+trait RpcHandler[Eff[_], Command, Response] {
+  import cats.data.Kleisli
+
+  /** the type of a Post payload, only used for Post calls */
+  type PostPayload
+
+  /** describes the effectful execution of a GET http call */
+  def getQuery: Kleisli[Eff, Command, Response]
+
+  /** describes the effectful execution of a POST http call */
+  def postQuery: Kleisli[Eff, (Command, Option[PostPayload]), Response]
+
+}
+
+/** Provides useful functions to interact with Rpc Nodes*/
+object RpcHandler {
+
+  /** An alias to express explicit type requirements including the internal Payload type of the RpcHandler */
+  type Aux[Eff[_], Command, Response, Payload] = RpcHandler[Eff, Command, Response] { type PostPayload = Payload }
+
+  /** Factor method based on an implicit instance being available in scope */
+  def apply[Eff[_], Command, Response, PostPayload](implicit rpc: Aux[Eff, Command, Response, PostPayload]) = rpc
+
+  /** A static call that uses the implicit `RpcHandler` instance available for the expected parameter types */
+  def runGet[Eff[_], Command, Response](
+      command: Command
+  )(
+      implicit
+      rpc: RpcHandler[Eff, Command, Response]
+  ): Eff[Response] =
+    rpc.getQuery.run(command)
+
+  /** A static call that uses the implicit `RpcHandler` instance available for the expected parameter types */
+  def runPost[Eff[_], Command, Response, PostPayload](
+      command: Command,
+      payload: Option[PostPayload] = None
+  )(
+      implicit
+      rpc: Aux[Eff, Command, Response, PostPayload]
+  ): Eff[Response] =
+    rpc.postQuery.run((command, payload))
+
+  /** Provides a generic way to convert the effect-type of an rpc handler, if there's
```

Comment:
Is there a built-in functionK for Future ~> IO we're using, and this case is just for illustration?

Reply:
No built-in, and here it's just illustrative, but I added mine as an implicit in `EffectsUtil`.

There is an issue with using it like this, though, because of how Futures are strict in executing. It's also explained in `TezosRemotesInstances` definition of the rpc-handler for `IO`